### PR TITLE
Add Druid 30.0.0 upgrade notes

### PR DESCRIPTION
**Pull request by ektravel**
This PR adds the upgrade notes for Druid 30.0.0.

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

File: docs/release-info/upgrade-notes.md [hunk 1]
```diff
@@ -26,6 +26,83 @@ The upgrade notes assume that you are upgrading from the Druid version that imme
...
+
+Druid 30.0.0 includes a breaking change that restores the behavior for `targetDataSource` to its 28.0.0 and earlier state, different from Druid 29.0.0 and only 29.0.0. In 29.0.0, `targetDataSource` returns a JSON object that includes the datasource name. In all other versions, `targetDataSource` returns a string containing the name of the datasource.
+
+If you're upgrading from any version other than 29.0.0, there is no change in behavior.
+
+If you are upgrading from 29.0.0, this is an incompatible change.
+
+[#16004](https://github.com/apache/druid/pull/16004)
+
+#### Removed ZooKeeper-based segment loading
```
Comment (cryptoe) [inline docs/release-info/upgrade-notes.md:81]: We need to call this out in the main release notes as one of the first line items as well.

**Review by cryptoe — APPROVED** [review]
Upgrades notes LGTM!!

File: docs/release-info/upgrade-notes.md (hunk 1 above)
Comment (ektravel) [reply docs/release-info/upgrade-notes.md:81]: @cryptoe The change you requested is on PR https://github.com/apache/druid/pull/16526.

**Review by 317brian — APPROVED** [review]
Gracias

**Pull request merged**